### PR TITLE
refactor: make mq interface simpler.

### PR DESCRIPTION
**Pull request by Rachelint**
# Which issue does this PR close?

Closes #

# Rationale for this change
The race's checking should not be done in mq interface level, it should be done in high-level. So I refactor it to make it simpler.
 

# What changes are included in this PR?
Refactor the mq interface and its kafka implemetation.

# Are there any user-facing changes?
None.

# How does this change test
By new and exist ut.

### CODE REVIEW

File: components/message_queue/src/kafka/kafka_impl.rs [hunk 1]
```diff
@@ -249,22 +253,62 @@ impl MessageQueue for KafkaImpl {
             })?)
     }
 
-    async fn consume_all(&self, topic_name: &str) -> Result<KafkaConsumeIterator> {
-        info!("Need to consume all data in kafka topic:{}", topic_name);
+    async fn get_earliest_offset(&self, topic_name: &str) -> Result<Offset> {
```
Comment (ShiKaiWi) [inline components/message_queue/src/kafka/kafka_impl.rs:256]: How about naming it `fetch_earliest_offset`? `Get` seems a lightweight operation.

File: components/message_queue/src/kafka/kafka_impl.rs [hunk 2]
```diff
@@ -249,22 +253,62 @@ impl MessageQueue for KafkaImpl {
...
+        topic_client
+            .get_offset(OffsetAt::Earliest)
+            .await
+            .context(FetchOffset {
+                topic_name: topic_name.to_string(),
+                offset_type: OffsetType::EarliestOffset,
+            })
+    }
 
+    async fn get_high_watermark(&self, topic_name: &str) -> Result<Offset> {
```
Comment (ShiKaiWi) [inline components/message_queue/src/kafka/kafka_impl.rs:274]:
```suggestion
    async fn fetch_high_watermark(&self, topic_name: &str) -> Result<Offset> {
```

BTW, `get_earliest_offset` and `get_high_watermark` seems the same procedure except the different offsets.

File: components/message_queue/src/lib.rs [hunk 3]
```diff
@@ -50,5 +61,27 @@ pub struct MessageAndOffset {
...
+    async fn next_message(&mut self) -> Result<(MessageAndOffset, Offset), Self::Error>;
+}
+
+/// At which position shall the stream start.
+#[derive(Debug, Clone, Copy)]
+pub enum StartOffset {
+    /// At the earlist known offset.
+    ///
+    /// This might be larger than 0 if some records were already deleted due to
+    /// a retention policy or delete records.
```
Comment (ShiKaiWi) [inline components/message_queue/src/lib.rs:73]:
```suggestion
    /// a retention policy or delete operations.
```

File: components/message_queue/src/lib.rs [hunk 4]
```diff
@@ -50,5 +61,27 @@ pub struct MessageAndOffset {
...
+    Earliest,
+
+    /// At the latest known offset.
+    ///
+    /// This is helpful if you only want ot process new data.
```
Comment (ShiKaiWi) [inline components/message_queue/src/lib.rs:78]:
```suggestion
    /// This is helpful if you only want to process new data.
```

File: components/message_queue/src/lib.rs (hunk 3 above)
Comment (Rachelint) [reply components/message_queue/src/lib.rs]: Done.

File: components/message_queue/src/lib.rs (hunk 4 above)
Comment (Rachelint) [reply components/message_queue/src/lib.rs]: Done.

File: components/message_queue/src/kafka/kafka_impl.rs (hunk 2 above)
Comment (Rachelint) [reply components/message_queue/src/kafka/kafka_impl.rs]: Done.

**Review by ShiKaiWi — APPROVED** [review]
LGTM

**Pull request merged**